dpe/src/crypto.rs

```diff
@@ -44,8 +44,31 @@ impl Default for EcdsaPub {
     }
 }
 
+pub trait Hasher {
+    /// Adds a chunk to the running hash.
+    ///
+    /// # Arguments
+    ///
+    /// * `bytes` - Value to add to hash.
+    fn update(&mut self, bytes: &[u8]) -> Result<(), DpeErrorCode>;
+
+    /// Finish a running hash operation and return the result.
```

Comment:
Is the expectation that after finish is called, the hasher can be reused? Probably doesn't matter either way, just would be good to document the expected behavior.